### PR TITLE
12 14 add form functionality to challenge submission page

### PR DESCRIPTION
**Pull request by RedBlueCarrots**
Added a fair bit of view/submit functionality, thanks to SQLAlchemy
(also accidentally did 14 inside this fix too...)

-Firstly fixed up the login with SQLAlchemy (very small fix, not worth an issue)
-Made it so submitting a request  will lodge in the database
-Updated the view, submission, and submit pages to render dynamically
-Made it so, on the page view:
1. Anon user see just challenge info
2. Creator user OR if the challenge is finished, see all the submissions
3. Submitted user will see their own submission
4. Unsubmitted user will see submission page

In future, we should add a resubmit button, but for MVP this will do

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -31,31 +32,52 @@ def create():
...
+    print(challenge)
+    pathString = challenge["startArticle"] + "|"
     for i in submitForm.path.data:
-        print(i)
-    return render_template('view.html', form=form, submitForm=submitForm, submitted=True)
+        if i.strip() != "":
+            pathString += i + "|"
+    pathString += challenge["endArticle"]
+    createNewSubmission(current_user.id, challenge["id"], pathString, int(time.time()))
+    return view(int(submitForm.challenge_id.data))
```
Comment (DeltaO3) [inline app/routes.py:44]: This should be a redirect. bad practice and makes reloading the page re-send data and confuse the browser

**Review by DeltaO3 — CHANGES_REQUESTED** [review]
Looks mostly good to me, but theres an issue with the url when submitting. Commented it, should be a redirect

File: app/routes.py (hunk 1 above)
Comment (RedBlueCarrots) [reply app/routes.py]: Should be updated

**Review by DeltaO3 — APPROVED** [review]
LGTM!

**Pull request merged**